core/TypePtr.h

```diff
@@ -255,6 +255,18 @@ class TypePtr final {
 
     bool hasUntyped() const;
 
+    // This is a hacky, unprincipled method that simply looks for void inside the type at the top
+    // level (not going into AppliedTypes).
```

Comment:
Are there cases where this would not catch `void` (because it doesn't go into `AppliedType`s)? If so, can you write a test case for those?